config/config/src/Config.ts

```diff
@@ -133,6 +134,23 @@ export interface Config {
   useRunningStoreServer?: boolean
   workspaceConcurrency: number
   workspaceDir?: string
+  workspaceManifest?: {
```

Comment:
I think we just need to add the setting from workspace manifest to the rest of the setting. Why should the users of the configs be aware where the settings are coming from? Just add workspacePatterns to the Config object.

Reply:
That's a good point. I was attempting to follow the `rootProjectManifest` option below, which grouped settings under where it comes from. It sounds like that wasn't a good example to follow?

What I do like about the existing setup here is that it better models what can happen at runtime. Grouping options together like this makes it possible to declare many options that are all non-null/defined if any of them are non-null. This means we don't have to handle non-existent states at compile time. In this case, we know both the workspace dir and patterns are non-null if either are non-null.

In a future PR, we'd also know the `patterns` and `dir` if catalogs are defined. And that intuitively makes sense since catalogs are a feature that only work in a workspace.